### DISCUSSION TITLE
Support for OpenAI's cached input tokens

### DISCUSSION THREAD

**Discussion by anatolec** [langfuse/langfuse#3866] [Ideas]
### Describe the feature or potential improvement

Since Oct 1st 2024, OpenAI introduced a [caching feature](https://platform.openai.com/docs/guides/prompt-caching) where common prefixes of successive prompts are cached resulting in a cost and latency decrease. The `usage` field in OpenAI's completion API response contains the detail of the number of cached tokens.

It would be ideal that LangFuse takes into account this cost discount to reflect the real cost.

**Comment by marcklingen** [comment]
Thanks for raising this! This is work in progress as we overhaul the cost/usage tracking in Langfuse to be very flexible and adapt to new pricing models, cc @hassiebp who is working on this

**Comment by juan-abia** [comment]
Any news on this one @hassiebp?

**Reply by hassiebp** [reply]
Support for this is part of our V3 release in the coming weeks 👍🏾

**Reply by kesh-model-ml** [reply]
so this is available now?

**Reply by marcklingen** [reply]
Yes
